src/jltypes.c

```diff
@@ -739,6 +739,36 @@ static jl_value_t *intersect_typevar(jl_tvar_t *a, jl_value_t *b,
                 break;
             }
         }
+        if (jl_is_typevar(b)) {
+            for(i=0; i < penv->n; i+=2) {
+                if (penv->data[i] == b && !jl_is_typevar(penv->data[i+1])) {
+                    if (jl_types_equal((jl_value_t*)a, penv->data[i+1])) {
```

Comment:
This seems very odd to me, since `a` is a typevar. I would think `jl_types_equal` would always return false in this case.